src/editor/action-executor.ts

```diff
@@ -0,0 +1,382 @@
+                        }
+                    );
+                }
+
+                break;
+
+            case EditAction.DisplayEqualsSuggestion:
+                suggestions = [ConstructKeys.Equals, ConstructKeys.NotEquals, ConstructKeys.VariableAssignment];
+                suggestions = this.module.getValidInsertsFromSet(focusedNode, suggestions);
+
+                this.module.menuController.buildSingleLevelMenu(
+                    suggestions,
+                    Util.getInstance(this.module).constructActions,
+                    {
+                        left: selection.startColumn * this.module.editor.computeCharWidth(),
+                        top: selection.startLineNumber * this.module.editor.computeCharHeight(),
+                    }
+                );
+
+                break;
+
+            case EditAction.OpenValidInsertMenu:
+                if (!this.module.menuController.isMenuOpen()) {
+                    this.module.menuController.buildAvailableInsertsMenu(
+                        this.module.getAllValidInsertsList(focusedNode),
+                        Util.getInstance(this.module).constructActions,
+                        {
+                            left: selection.startColumn * this.module.editor.computeCharWidth(),
+                            top: selection.startLineNumber * this.module.editor.computeCharHeight(),
+                        }
+                    );
+                } else this.module.menuController.removeMenus();
+
+                break;
+
+            //TODO: Remove later
+            case EditAction.OpenValidInsertMenuSingleLevel:
+                if (!this.module.menuController.isMenuOpen()) {
+                    const suggestions = this.module.getAllValidInsertsList(focusedNode);
+                    this.module.menuController.buildSingleLevelConstructCategoryMenu(suggestions);
+                } else this.module.menuController.removeMenus();
+
+                break;
+
+            case EditAction.SelectMenuSuggestionAbove:
+                this.module.menuController.focusOptionAbove();
+
+                break;
+
+            case EditAction.SelectMenuSuggestionBelow:
+                this.module.menuController.focusOptionBelow();
+
+                break;
+
+            case EditAction.SelectMenuSuggestion:
+                this.module.menuController.selectFocusedOption();
+
+                break;
+
+            case EditAction.CloseValidInsertMenu:
+                this.module.menuController.removeMenus();
+
+                break;
+
+            case EditAction.OpenSubMenu:
+                this.module.menuController.openSubMenu();
+
+                break;
+
+            case EditAction.CloseSubMenu:
+                this.module.menuController.closeSubMenu();
+
+                break;
+
+        }
+
+        return preventDefaultEvent;
+    }
+
+    
+    private validateIdentifier(context: Context, identifierText: string) {
```

Comment:
Can probably move this to the validator now.

Reply:
I wanted to do this, however, as it was calling some functions for the notification, I decided to stop and see if we want to separate them or not.

Reply:
I see. Ok let's just merge it this way for now and then deal with that later.